lib/secure_random.ex

```diff
@@ -108,22 +109,23 @@ defmodule SecureRandom do
   end
 
   defp encode(<<u0::32, u1::16, u2::16, u3::16, u4::48>>) do
-    hex_pad(u0, 8) <> "-" <>
-    hex_pad(u1, 4) <> "-" <>
-    hex_pad(u2, 4) <> "-" <>
-    hex_pad(u3, 4) <> "-" <>
-    hex_pad(u4, 12)
+    hex_pad(u0, 8) <>
```

Comment:
(opinion) To this reader of the code this is not a good change. Perhaps this is really better expressed something like:

```elixir
  # Warning: the bit sizes of the binary match are all 4 times the field widths
  # in the format string. If you change one you’ll need to change the other.
  def encode(<<u0::32, u1::16, u2::16, u3::16, u4::48>>) do
    "~8.16.0b-~4.16.0b-~4.16.0b-~4.16.0b-~12.16.0b"
    |> :io_lib.format([u0, u1, u2, u3, u4])
    |> to_string()
  end
```

and then we can get rid of `hex_pad/2` and `lower/2`.

Reply:
I think that there's a different way to do this, but this code was imported from an existing PR and only slightly changed. If I decide that I want to fork this and release it under a different name, I’ll probably look at a few other changes than just this.